common/perf-test-core/src/main/java/com/azure/perf/test/core/PerfStressProgram.java

```diff
@@ -249,7 +258,19 @@ public static void runTests(PerfTestBase<?>[] tests, boolean sync, int parallel,
                 forkJoinPool.invokeAll(operations);
 
                 forkJoinPool.awaitQuiescence(durationSeconds + 1, TimeUnit.SECONDS);
-            } else {
+            } else if (completableFuture) {
+                for (PerfTestBase<?> test : tests) {
+                    test.runAllAsyncWithCompletableFuture(endNanoTime).get();
+                }
+            } else if (executorService) {
+                for (PerfTestBase<?> test : tests) {
+                    test.runAllAsyncWithExecutorService(endNanoTime);
+                }
+            } else if (virtualThread) {
+                for (PerfTestBase<?> test : tests) {
+                    test.runAllAsyncWithVirtualThread(endNanoTime);
```

Comment:
Overall, for the three new scenarios added, should the logic still follow what is being done for synchronous testing, just that we're changing the mechanism making performance tracked calls